ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md

```diff
@@ -0,0 +1,111 @@
+3. **Descriptions:**
+    - Only add descriptions to the keys when they provide additional context beyond the key name.
+    - Refer Arista documentation for description content.
+    - Ensure all descriptions end with punctuation.
+    - Highlight the key names in description, like - `<key_name>`.
+4. **Type Conversion:**
+    - Add `convert_types: [str]` for `type: int` keys.
+    - Add `convert_types: [int]` for `type: str` if it can be numeric.
+5. **Defaults:** Avoid using `default` in eos_cli_config_gen.
+6. **Min/Max:** Specify min/max values in the schema if they are defined in the EOS CLI. Make sure to check on different hardware platforms.
+7. **Valid_Values:** Specify valid options in the schema as per the EOS CLI and ensure compatibility across different hardware platforms.
+
+### Creating Jinja2 Templates
+
+Edit the appropriate jinja2 templates in `pyavd/_eos_cli_config_gen/j2templates/eos` and `pyavd/_eos_cli_config_gen/j2templates/documentation` to generate the desired configuration and documentation.
+
+Add a new jinja2 template if adding a top-level feature, also modify the `pyavd/_eos_cli_config_gen/j2templates/eos-intended-config.j2` and `pyavd/_eos_cli_config_gen/j2templates/eos-device-documentation.j2` to add these new templates, trying to respect the order in the EOS CLI.
+
+#### Jinja2 Templates Guidelines
+
+1. **Code Indentation:** Keep less indented code to improve readability.
+2. **Variable Naming:** Use meaningful variable names. Avoid short variables like `ei` for `ethernet_interface`
+3. **Use AVD filters:** Use AVD filters for code optimization - [AVD Filters](https://avd.arista.com/5.0/docs/plugins/Filter_plugins/add_md_toc.html).
+4. **Natural Sorting:** Use `arista.avd.natural_sort` for sorting the `for loops` after checking on EOS CLI.
+5. **Defined Checks:**
+    - Avoid `arista.avd.defined` check for parent keys when directly checking for child keys.
+    - Avoid `arista.avd.defined` check for primary and required keys.
+    - Avoid`arista.avd.defined` check when using `arista.avd.default()` and `arista.avd.natural_sort` filters.
+6. **Password Security:** Avoid displaying passwords in the documentation template and use the `arista.avd.hide_passwords` filter to hide it.
+7. **Config Order:** Ensure the order and indentation of configuration matches EOS CLI.
+8. **Exclamation Marks:** Place exclamation marks `!` correctly as per the EOS running-config.
+9. Along with EOS config template, update the documentation template as well (if required).
+10. Implement only commands visible in `show running-config all` of the EOS device. We should not hide config if given by the user.
+11. Validate the template using j2lint tool, run `pre-commit run j2lint --all`.
+
+### Build Schemas and Documentation
+
+Run `pre-commit run schemas --all` to re-generate the eos_cli_config_gen schema with any modifications. This command should be executed every time the schema is changed, even if only a description is updated.
```

Comment:
Starting AVD 5.1 - it will happen automaticaly when running molecule but will be required if trying with an external repo

Reply:
basically if you run from source (which would happen in molecule / if you run the collection from its github repo) you would not need to do this when running through Ansible. When using pyavd you need to.

I think it is fair to state something like "When developing a feature you should be running from source using Ansible (either via molecule or from a test repo) and so the schemas and templates should be auto-recompiled during the "Verify Requirements" step as per https://avd.arista.com/stable/docs/contribution/development-tooling.html#running-from-source. When running pyavd or if you need to recompile for another reason use `pre-commit run schemas --all` and / or `pre-commit run templates --all`."

Reply:
Updated!